projects/demo/src/modules/customization/viewport/examples/2/portal-host.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    Inject,
+    INJECTOR,
+    Injector,
+} from '@angular/core';
+import {
+    AbstractTuiPortalHostComponent,
+    AbstractTuiPortalService,
+    TuiDropdownPortalService,
+} from '@taiga-ui/cdk';
+import {tuiAsViewport, TuiRectAccessor} from '@taiga-ui/core';
+
+@Component({
+    selector: `[portalHost]`,
+    template: `
+        <ng-content></ng-content>
+        <ng-container #viewContainer></ng-container>
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {provide: AbstractTuiPortalService, useExisting: TuiDropdownPortalService},
+        {
+            provide: AbstractTuiPortalHostComponent,
+            useExisting: PortalHost,
+        },
+        tuiAsViewport(PortalHost),
+    ],
+})
+export class PortalHost
+    extends AbstractTuiPortalHostComponent
+    implements TuiRectAccessor
+{
+    readonly type = `viewport`;
+
+    constructor(
```

Comment:
Do we need `constructor` here? I think once you've made elementRef `protected` in the abstract class you can just drop it.